### ISSUE TITLE
Expanded Listeners

### ISSUE THREAD

**Issue by rmrhz** [super3/IRC-Bot#34]
Hello guys, I wonder if this idea is ok or not but recently I've been doing some changes to enable the Listener to have _custom triggers_ and _custom actions_

![168](https://f.cloud.github.com/assets/4339240/551617/b6d6a51a-c32e-11e2-9fdf-2a3e1bfb33b4.png)
_original code at the bottom portion_

![169](https://f.cloud.github.com/assets/4339240/551622/e1b66284-c32e-11e2-96bc-9022d9678293.png)

I know by doing this, this will defeat the purpose of Commands if it may be possible to combine them it will be great as it will be more extendable and organized in such a way, I don't need to switch files every once in a while whenever I'm editing a command.

This will also let me customize which part of the arguments to be used, it can be the user[1] the command [2] the parameters [3][4][5] so on and so forth.

I'm also planning to port it with database functions as for the logging and other ideas that my head is making.

Cheers!

**Comment by Yoshi2889** [comment]
I'm thinking of adding certain new types of listeners at a later stage and making them overall more powerful - I've set a reminder to tell you more tomorrow.

**Comment by rupaheizu** [comment]
Alright got it

**Comment by Yoshi2889** [comment]
Right. So in my own build of the bot I added a new type of listeners that scan the message for keywords. They work just like before, they just define a getMessageKeywords() function instead of getKeywords().

As for ways to make them more powerful, I'd suggest adding regex (although that could get very slow) and multiple keywords.

They're not major additions. Maybe other types of listeners (e.g. per-user) would be nice too.
